assets/src/pages/room/hooks/useMedia.tsx

```diff
@@ -0,0 +1,199 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+
+export type UseMediaResult = State & Api;
+
+type State = {
+  isError: boolean;
+  isSuccess: boolean;
+  stream?: MediaStream;
+  isEnabled: boolean;
+};
+
+export type Api = {
```

Comment:
Such name is very vague and enigmatic. Can you use something more specific? Like `MediaApi` or something